### PR TITLE
[generator] Avoid some unrequired loops over single usage attributes on members

### PR DESCRIPTION
**Pull request by spouliot**

### CODE REVIEW

File: src/generator.cs [hunk 1]
```diff
@@ -5191,10 +5191,9 @@ void PrintMethodAttributes (MemberInformation minfo)
 			PrintPlatformAttributes (minfo.method.DeclaringType);
 		}
 
-		foreach (var di in AttributeManager.GetCustomAttributes<DesignatedInitializerAttribute> (mi)) {
+		// in theory we could check for `minfo.is_ctor` but some manual bindings are using methods for `init*`
+		if (AttributeManager.GetCustomAttribute<DesignatedInitializerAttribute> (mi) != null)
```
Comment (rolfbjarne) [inline src/generator.cs]:
```suggestion
		if (AttributeManager.HasAttribute<DesignatedInitializerAttribute> (mi))
```

File: src/generator.cs [hunk 2]
```diff
@@ -6409,10 +6408,8 @@ public void Generate (Type type)
 					if (external) {
 						if (!disable_default_ctor) {
 							GeneratedCode (sw, 2);
-							foreach (var ta in AttributeManager.GetCustomAttributes<DesignatedDefaultCtorAttribute> (type)) {
+							if (AttributeManager.GetCustomAttributes<DesignatedDefaultCtorAttribute> (type) != null)
```
Comment (rolfbjarne) [inline src/generator.cs]:
```suggestion
							if (AttributeManager.HasAttribute<DesignatedDefaultCtorAttribute> (type))
```

File: src/generator.cs [hunk 3]
```diff
@@ -6435,10 +6432,8 @@ public void Generate (Type type)
 					} else {
 						if (!disable_default_ctor) {
 							GeneratedCode (sw, 2);
-							foreach (var ta in AttributeManager.GetCustomAttributes<DesignatedDefaultCtorAttribute> (type)) {
-								sw.WriteLine ("\t\t[DesignatedInitializer]");
-								break;
-							}
+							if (AttributeManager.GetCustomAttributes<DesignatedDefaultCtorAttribute> (type) != null)
```
Comment (rolfbjarne) [inline src/generator.cs]:
```suggestion
							if (AttributeManager.HasAttribute<DesignatedDefaultCtorAttribute> (type))
```

**Comment by spouliot** [comment]
Weird, I looked for it but did not find it... typo probably, since I see it now :)

**Comment by rolfbjarne** [comment]
> _edit_ I probably looked for `HasCustomAttribute`, not `HasAttribute`

You could always add a comment to `HasAttribute` :smile:

```csharp
// This is the function you're looking for if you're looking for HasCustomAttribute
```

**Comment by spouliot** [comment]
16x HE0038 -> known issue on watchOS https://github.com/xamarin/maccore/issues/581
16x iOS 32bits iOS timeouts -> https://github.com/xamarin/maccore/issues/2069
both related to 32bits execution issues on some bots like `xam-mac-mini-38`

**Comment by spouliot** [comment]
a single HE0038 -> known issue on watchOS xamarin/maccore#581

**Pull request merged**